firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/ImpressionStorageClient.java

```diff
@@ -96,4 +98,39 @@ private void clearInMemCache() {
         .map(CampaignImpression::getCampaignId)
         .contains(campaignId);
   }
+
+  /**
+   * Clears impressions for all campaigns found in the provided {@link
+   * FetchEligibleCampaignsResponse}
```

Comment:
Lets add why we do this? (something like: this is required because we trust the server to tell us which campaigns can be re-impressed to honor frequency limits)?